include/dali/core/expand_dims.h

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_CORE_EXPAND_DIMS_H_
+#define DALI_CORE_EXPAND_DIMS_H_
+
+#include "dali/core/tensor_shape.h"
+
+namespace dali {
+
+/**
+ * @brief  Inserts new dimensions with extent one to the new shape as needed to match a given number
+ of dimensions
+
+ *
+ * @param sh input shape
+ * @param new_dims_position Position where the new dimensions are inserted
+ * @param dynamic_out_dim Dynamic number of dimensions in the output (relevant if static_out_ndim ==
+ DynamicDimensions)
+ */
+template <int static_out_ndim = DynamicDimensions, int static_in_ndim = DynamicDimensions>
+TensorListShape<static_out_ndim> expand_dims(const TensorListShape<static_in_ndim> &sh,
+                                             int new_dims_position = 0, int dynamic_out_dim = -1) {
```

Comment:
To consider: add a variant that fills an output parameter and then implement this as:
```C++
TensorListShape<static_out_ndim> expand_dims(const TensorListShape<static_in_ndim> &sh,
                                             int new_dims_position = 0, int dynamic_out_dim = -1) {
  TensorListShape<static_out_ndim> new_sh;
  expand_dims(new_sh, sh, new_dims_position, dynamic_out_dim);
  return new_sh;
}
```